simulator/core/perf_sim.cpp

```diff
@@ -261,23 +261,23 @@ void PerfMIPS::clock_memory( int cycle)
         return;
     }
 
-    /* acquiring real information */
-    bool actually_taken = instr.is_jump_taken();
-    Addr real_target = instr.get_new_PC();
-
-    /* branch misprediction unit */
-    if ( instr.is_misprediction())
-    {
-        /* updating BTB */
+    if (instr.isJump()) {
```

Comment:
Congrats, you have stolen two points from this year students. I was going to ask one of them to remove this bug. Have I asked you about it? No I haven't! Anyway, do not revert your patch, it is exposed already to everyone.

Reply:
1) Sorry, I didn't know that this issue is already present in your list.
2) It's not just about changing this one line, it was the flaw with whole BPU logic itself.